vpc/service/assign_addresses_v3.go

```diff
@@ -803,33 +803,19 @@ func assignElasticAddressesBasedOnGroupName(ctx context.Context, tx *sql.Tx, ses
 		return nil, err
 	}
 
-	row := tx.QueryRowContext(ctx, `
-SELECT allocation_id, public_ip
-FROM elastic_ips
-WHERE account_id = $1
-  AND allocation_id NOT IN
-    (SELECT elastic_ip_allocation_id
-     FROM elastic_ip_attachments)
-  AND network_border_group = $2
-  AND tags->>'titus_vpc_pool' = $3
-LIMIT 1
-FOR
-UPDATE of elastic_ips
-`, aws.StringValue(branchENI.OwnerId), borderGroup, groupName)
-	var allocationID, publicIP string
-	err = row.Scan(&allocationID, &publicIP)
-	if err == sql.ErrNoRows {
-		err = fmt.Errorf("No EIP in group %s free in network border group %s", groupName, borderGroup)
-		span.SetStatus(traceStatusFromError(err))
-		return nil, err
-	} else if err != nil {
-		err = errors.Wrap(err, "Cannot query for free elastic IPs")
+	elasticAddress, err := db.GetAvailableElasticAddressByGroupAndLock(ctx, tx, aws.StringValue(branchENI.OwnerId), borderGroup, groupName)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			err = fmt.Errorf("no EIP in group %s free in network border group %s", groupName, borderGroup)
+		} else if err != nil {
```

Comment:
This error handling here is a bit of a departure from the original code.
At this line, err is already != nil, we'll always hit it.
I guess it is not technically incorrect.

Reply:
I moved `err == sql.ErrNoRows` check to under `err != nil` check so that it's clear that when `err != nil`, we always return from here. You're right about the redundant `else if`. Will change it to `else` only.